modules/rosa-hcp-sts-creating-a-cluster-ext-auth-provider-cli.adoc

```diff
@@ -0,0 +1,136 @@
+====
+
+This configuration uses Microsoft Entra ID to authenticate your credentials. The Entra ID values are:
+
+[cols="1,1", options="header"]
+|===
+
+|CLI Flag
+|Description
+
+|`--cluster` 
+|The name or the ID of your cluster
+
+|`--name`
+|A name that is used to refer to the external authentication provider.
+
+|`--console-client-secret`
+|This string is the client secret that is used to associate your account with the application. If you do not include the client secret, this command uses a public OIDC OAuthClient.
+
+|`--issuer-audiences` 
+|This comma-separated list of client audiences. In Microsoft Entra ID, this is called the audience ID.
+
+|`--issuer-url`
+|The URL of the token issuer.
+
+|`--claim-mapping-username-claim` 
+|The name of the claim that should be used to construct user names for the cluster identity.
+
+|`--claim-mapping-groups-claim`
+|The name of the claim that should be used to construct group names for the cluster identity.
+
+|===
+
+.Procedure
+
+. Use one of the following commands to create your external authentication provider:
+
+* To use the interactive command interface, run the following commands:
++
+[source,terminal]
+----
+$ rosa create external-auth-provider -c <cluster_name>
+----
++
+This command starts an interactive CLI process:
++
+--
+[source,terminal]
+----
+I: Enabling interactive mode
+? Name: <.>
+? Issuer audiences: <.>
+? The serving url of the token issuer: <.>
+? CA file path (optional): <.>
+? Claim mapping username: <.>
+? Claim mapping groups: <.>
+? Claim validation rule (optional): <.>
+? Console client id (optional): <.>
+----
+<.> The name of your external authentication provider. This name should be a lower-case with numbers and dashes.
+<.> The issuer audiences field requires you to provide the audience IDs that this authentication provider issues tokens for.
+<.> This field uses the issuer's URL that serves the token.
+<.> Optional: This field takes the path of a certificate file to use when making requests.
+<.> The name of the claim that is used to construct the user names for cluster identity, such as using `email`.
+<.> This field specifies how to transform the ID token into a cluster identity, such as using `groups`.
+<.> Optional: These are rules that help validate token claims which authenticate your users. This field should be formatted as `:<required_value>`.
+<.> Optional: This field is the application or client ID that your app registration uses for the console.
+--
+
+* You can include the required IDs to create your external authentication provider with the following command:
++
+[source,terminal]
+----
+rosa create external-auth-provider -c <cluster_id> --name <provider_name> --issuer-audiences <audience_id> --issuer-url <issuing id> --claim-mapping-username-claim email --claim-mapping-groups-claim groups
```

Comment:
I would add the example that uses the console as well.

Something like this example:

```
rosa create external-auth-provider \
--cluster=lponce-local-01 \
--name=m-entra-id \
--issuer-url=https://login.microsoftonline.com/fa5d3dd8-b8ec-4407-a55c-ced639f1c8c5/v2.0 \
--issuer-audiences=a9464024-b142-4bdf-86c0-a153109cdb14,8a769b34-13c9-4f5b-9933-ec439700ec67 \
--claim-mapping-username-claim=email \
--claim-mapping-groups-claim=groups \
--console-client-id=8a769b34-13c9-4f5b-9933-ec439700ec67 \
--console-client-secret=<SECRET>
```